### PR TITLE
fix(localize): enable whitespace preservation marker in XLIFF files

### LINKED ISSUE #38679: Space preservation attribute not added to XLF file
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/localize^10.1.0-rc.0

### Description

In ts code I use i18n:
``` typescript
title = $localize`hello\nworld` // tooltip should be shown in two lines
```
Then I call `localize-extract` and get xlf file without `xml:space="preserve"` attribute:
``` xml
  ...
  <file source-language="en" datatype="plaintext" original="ng2.template">
  ...
      <trans-unit id="1122334455667788990" datatype="html">
        <source>Hello
world</source>
  ...
````
When I add translation in CAT tool (OmegaT), line break is gone.

## 🌍  Your Environment

**Angular Version:**
```
10.1.0-rc.0
```

### PR DESCRIPTION
**Pull request by petebacondarwin**
Whitespace can be relevant in extracted XLIFF translation files.
Some i18n tools - e.g. CAT tool (OmegaT) - will reformat
the file to collapse whitespace if there is no indication to tell it
not to.

This commit adds the ability to specify "format options" that are passed
to the translation file serializer. The XLIFF 1.2 and 2.0 seralizers have
been updated to accept `{"xml:space":"preserve"}` format option which will
by added to the `<file>` element in the serialized translation file during
extraction.

Fixes #38679

### CODE REVIEW

File: packages/localize/src/tools/src/extract/translation_files/format_options.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+export type FormatOptions = Record<string, string>;
+export type ValidOption = [string, string[]];
```
Comment (AndrewKushnir) [inline packages/localize/src/tools/src/extract/translation_files/format_options.ts:10]: We can probably use names tuples here:
```suggestion
export type ValidOption = [key: string, value: string[]];
```

File: packages/localize/src/tools/src/extract/translation_files/xliff1_translation_serializer.ts [hunk 2]
```diff
@@ -24,9 +25,12 @@ const LEGACY_XLIFF_MESSAGE_LENGTH = 40;
  * @see Xliff1TranslationParser
  */
 export class Xliff1TranslationSerializer implements TranslationSerializer {
+  static validFormatOptions = new Map<string, string[]>([['xml:space', ['preserve']]]);
```
Comment (AndrewKushnir) [inline packages/localize/src/tools/src/extract/translation_files/xliff1_translation_serializer.ts:28]: Looks like this is unused.

File: packages/localize/src/tools/src/extract/translation_files/xliff1_translation_serializer.ts [hunk 3]
```diff
@@ -24,9 +25,12 @@ const LEGACY_XLIFF_MESSAGE_LENGTH = 40;
...
   constructor(
-      private sourceLocale: string, private basePath: AbsoluteFsPath,
-      private useLegacyIds: boolean) {}
+      private sourceLocale: string, private basePath: AbsoluteFsPath, private useLegacyIds: boolean,
+      private formatOptions: Record<string, string>) {
```
Comment (AndrewKushnir) [inline packages/localize/src/tools/src/extract/translation_files/xliff1_translation_serializer.ts:31]: We can probably use `FormatOptions` type here:
```suggestion
      private formatOptions: FormatOptions) {
```

File: packages/localize/src/tools/src/extract/translation_files/xliff2_translation_serializer.ts [hunk 4]
```diff
@@ -25,8 +26,10 @@ const MAX_LEGACY_XLIFF_2_MESSAGE_LENGTH = 20;
 export class Xliff2TranslationSerializer implements TranslationSerializer {
   private currentPlaceholderId = 0;
   constructor(
-      private sourceLocale: string, private basePath: AbsoluteFsPath,
-      private useLegacyIds: boolean) {}
+      private sourceLocale: string, private basePath: AbsoluteFsPath, private useLegacyIds: boolean,
+      private formatOptions: Record<string, string>) {
```
Comment (AndrewKushnir) [inline packages/localize/src/tools/src/extract/translation_files/xliff2_translation_serializer.ts:30]:
```suggestion
      private formatOptions: FormatOptions) {
```

**Review by AndrewKushnir — APPROVED** [review]
LGTM, thanks @petebacondarwin 👍 

(just left a couple minor comments)

File: packages/localize/src/tools/src/extract/translation_files/format_options.ts (hunk 1 above)
Comment (petebacondarwin) [reply packages/localize/src/tools/src/extract/translation_files/format_options.ts:10]: Ooh! I didn't even know that named tuples were a thing!

**Review by josephperrott — APPROVED** [review]
LGTM

**Comment by petebacondarwin** [comment]
Failure is due to size change on master.

**Pull request closed without merging**